planner/core/find_best_task.go

```diff
@@ -759,9 +772,29 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty, planCounter
 		}
 	}
 
+	if safeUpdateMode && numRangeScanPlans == 0 {
```

Comment:
Should we just check the `best` plan instead of every plan? The `range scan` plan may not be chosen as the chosen one.

Reply:
If the optimizer chooses a full-scan as the best plan, but actually some other range-scans can be used, we shouldn't return the error in this case.